### PR TITLE
WIP Storage for current Setup

### PR DESCRIPTION
**Pull request by mRs-**
In https://github.com/BenchR267/Get-Schwifty/issues/9 a storage was requested. Here my current, and not really clean implementation :)

It's not finished yet. Storing is working but you can't select a new one.

Storage is working like apple charm (without any user to do things about this, its just save the things).

### CODE REVIEW

**Comment by kiliankoe** [comment]
Mobile GitHub stinks, I can't just add comments without it immediately being a review, meh...

**Comment by mRs-** [comment]
just do it ^^

**Comment by mRs-** [comment]
The implementation is now working. But i have problem using the current setup with the PageViewController. 

@BenchR267 you must decide how this should work.

**Comment by BenchR267** [comment]
Thanks for the contribution, looks really good! I will take care of integrating it in the UI and change some things :)

**Comment by BenchR267** [comment]
I changed a few things and integrated it in the Get Schwifty UI. I would like to finish this feature in this branch before merging it. Currently open are the following things:

- [ ] name should be changable
- [ ] saving files should be enhanced (when is it done, when is asked for the name etc)

~~list view should be a collection view instead of table view~~
~~item deletion is implemented but gesture is disturbed by page view controller -> using collection view should change this issue~~
~~(optional) it would be extremely awesome, if the project overview is not a few that's side by side to the editor but appears after zooming out~~

If you think this is too much for this pull request we could also open a feature branch so everyone who would like to help could open pull requests for smaller parts before merging the whole feature branch into master.

File: GetSchwifty/View Controllers/ProjectTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    public func newSchwifty() {
+        self.delegate?.projectTableViewControllerDidSelect(schwifty: Schwifty(source: ""))
+    }
+    
+    // MARK: - Table view data source
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
```
Comment (mRs-) [inline GetSchwifty/View Controllers/ProjectTableViewController.swift]: Maybe we need to dispatch this dynamically, the section could be empty

**Comment by mRs-** [comment]
looks good to me but imho we should move the collection view things to another issue, because it's not a blocker for this issue.

Naming and other stuffs are required imho

**Comment by BenchR267** [comment]
Makes sense! I updated the comment above and will create related issues later :)

File: GetSchwifty/View Controllers/ProjectTableViewController.swift (hunk 1 above)
Comment (BenchR267) [reply GetSchwifty/View Controllers/ProjectTableViewController.swift]: If we would like to hide the section completly if empty, yes. But this is not absolutely needed in my eyes.

File: .gitignore [hunk 2]
```diff
@@ -1,6 +1,8 @@
 
 # Created by https://www.gitignore.io/api/Xcode
 
+.DS_Store
```
Comment (BenchR267) [inline .gitignore:4]: 👍

File: GetSchwifty/Archiver/Schwifty.swift [hunk 3]
```diff
@@ -16,11 +16,11 @@ private let dateFormatter: DateFormatter = {
 }()
 
 class Schwifty {
-    
+
     private enum CodingKeys: String {
-        case Date, Name, Source, Temporary
+        case date, name, source, temporary
```
Comment (BenchR267) [inline GetSchwifty/Archiver/Schwifty.swift]: It's still not in my head to write those lower case :D

Comment (mRs-) [reply GetSchwifty/Archiver/Schwifty.swift]: same here :/

**Comment by mRs-** [comment]
Will finish this today in the train to UIKonf

**Comment by BenchR267** [comment]
Have a good ride, see you today! 🙂

File: GetSchwifty/Archiver/Schwifty.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+private let dateFormatter: DateFormatter = {
+    let d = DateFormatter()
+    d.timeStyle = .long
+    d.dateStyle = .long
+    return d
+}()
+
+class Schwifty {
+
+    private enum CodingKeys: String {
```
Comment (BenchR267) [inline GetSchwifty/Archiver/Schwifty.swift:20]: This is actually pretty near at the implementation in Swift 4 :blush:

**Comment by BenchR267** [comment]
@mRs- should I close this one? I think no one will continue working on this…

**Comment by mRs-** [comment]
you can close it, when I got time I will reopen it.

**Pull request closed without merging**